cypress-tests/README.md

```diff
@@ -180,103 +225,169 @@ To add a new connector for testing with Hyperswitch, follow these steps:
+beforeEach(() => {
+  cy.log("Current state:", JSON.stringify(globalState.data));
 });
 ```
 
-You can create similar scenarios by calling other functions defined in `commands.js`. These functions interact with utility files like `<connector_name>.js` and include necessary assertions to support various connector scenarios.
+### 5. Hooks
+
+- If the `globalState` object does not contain latest data, it must be due to the hooks not being executed in the correct order
+- Add `cy.log(globalState)` to the test case to verify the data in the `globalState` object
+
+> [!NOTE]
+> Refer to the Cypress's official documentation for more information on hooks and their execution order [here](https://docs.cypress.io/app/core-concepts/writing-and-organizing-tests#Hooks).
+
+### 6. Tasks
+
+- Use `cy.task` to interact with the Node.js environment
+- Task can only be used in `support` files and `spec` files. Using them in files outside these directories will result in unexpected behavior or errors like abrupt termination of the test suite
+
+````javascript
+
+## Linting
+
+To run the formatting and lint checks, execute the following command:
+
+```shell
+# Format the code
+npm run format
+
+# Check the formatting
+npm run format:check
 
-### Debugging
+# Lint the code. This wont fix the logic issues, unused imports or variables
+npm run lint -- --fix
+````
 
-It is recommended to run `npm run cypress` while developing new test cases to debug and verify as it opens the Cypress UI allowing the developer to run individual tests. This also opens up the possibility to to view the test execution in real-time and debug any issues that may arise by viewing the request and response payloads directly.
+## Best Practices
 
-If, for any reason, the `globalState` object does not contain latest data, it must be due to the hooks not being executed in the correct order. In such cases, it is recommended to add `cy.log(globalState)` to the test case to verify the data in the `globalState` object.
-Please refer to the Cypress's official documentation for more information on hooks and their execution order [here](https://docs.cypress.io/app/core-concepts/writing-and-organizing-tests#Hooks).
+1. Use the global state for sharing data between tests
+2. Implement proper error handling
+3. Use appropriate wait strategies
+4. Maintain test independence
+5. Follow the existing folder structure
+6. Document connector-specific behaviors
+7. Use descriptive test and variable names
+8. Use custom commands for repetitive tasks
+9. Use `cy.log` for debugging and do not use `console.log`
 
 ## Additional Resources
 
-For more information on using Cypress and writing effective tests, refer to the official Cypress documentation: [Cypress Documentation](https://docs.cypress.io/)
+- [Cypress Documentation](https://docs.cypress.io/)
+- [API Testing Best Practices](https://docs.cypress.io/guides/end-to-end-testing/api-testing)
+- [Hyperswitch API Documentation](https://hyperswitch.io/docs)
```

Comment:
Do we wanna include the link to the API reference here instead?

Reply:
this was added as an optional reference so that people who want to see how the request and response would be for a specific feature before adding a test might find it helpful. can be removed.